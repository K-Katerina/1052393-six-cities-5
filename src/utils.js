import moment from "moment";
import {MAX_NEAR_PLACES, TypeCards} from "./const";

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getNearOffers = (offers) => {
  return offers.slice(0, MAX_NEAR_PLACES);
};

export const getOffersForCity = (city, offers) => {
  return offers.filter((offer) => offer.location === city.name);
};

export const sortOffers = (offers, sortType) => {
  switch (sortType) {
    case `PRICE_TO_HIGH`: return offers.sort((a, b) => a.costPerNight - b.costPerNight).slice();
    case `PRICE_TO_LOW`: return offers.sort((a, b) => b.costPerNight - a.costPerNight).slice();
    case `BY_RATING`: return offers.sort((a, b) => b.rating - a.rating).slice();
    default: return offers.slice();
  }
};

export const getRating = (rating) => Math.round(rating / 5 * 100) + `%`;

export const getDate = (date) => moment(date).format(`MMMM YYYY`);

export const groupOffersByLocation = (offers) => {
  const map = new Map();
  offers.filter((offer) => offer.isFavorite).forEach((offer) => {
    const location = offer.location;
    map.set(location, map.get(location) || []);
    map.get(location).push(offer);
  });
  return map;
};


export const getStyleForCard = (cardType) => {
  switch (cardType) {
    case TypeCards.CITIES:
      return {
        className: `cities`,
        width: 260,
        height: 200
      };
    case TypeCards.NEAR_PLACES:
      return {
        className: `near-places`,
        width: 260,
        height: 200
      };
    case TypeCards.FAVORITES:
      return {
        className: `favorites`,
        width: 150,
        height: 110
      };
    default:
      return new Error(`Type "${cardType}" not found`);
  }
};

export const capitalizeWord = (word) => {
  return word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();
};
