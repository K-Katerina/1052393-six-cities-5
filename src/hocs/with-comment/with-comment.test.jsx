import React from "react";
import renderer from "react-test-renderer";
import withComment from "./with-comment";
import PropTypes from "prop-types";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import configureStore from "redux-mock-store";
import {makeInitialStateMock} from "../../utils";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired
};

const MockComponentWrapped = withComment(MockComponent);

it(`withComment is render correctly`, () => {
  const store = configureStore()(makeInitialStateMock());

  const tree = renderer.create((
    <Provider store={store}>
      <BrowserRouter>
        <MockComponentWrapped
          postReview={()=>{}}
          id={1}
        >
          <React.Fragment />
        </MockComponentWrapped>
      </BrowserRouter>
    </Provider>
  )).toJSON();
  expect(tree).toMatchSnapshot();
});
