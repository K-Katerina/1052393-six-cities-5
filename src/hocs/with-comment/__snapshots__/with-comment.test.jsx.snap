// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withComment is render correctly 1`] = `<div />`;
